src/AzModuleInstaller.ts

```diff
@@ -0,0 +1,159 @@
+import * as core from '@actions/core';
+import * as tc from '@actions/tool-cache';
+import * as os from 'os';
+import { ArchiveTools } from './Utilities/ArchiveTools';
+import FileUtils from './Utilities/FileUtils';
+import Utils from './Utilities/Utils';
+
+interface InstallResult {
+    moduleSource: string;
+    isInstalled: boolean;
+}
+
+export class AzModuleInstaller {
+    private version: string;
+    private githubAuth: string;
+    private moduleContainerPath: string;
```

Comment:
imo azModuleRoot seems like a more descriptive name.

Reply:
Renamed to moduleRoot. Since class name clear we are talking about Az so no prepending that.